src/task.js

```diff
@@ -34,18 +34,20 @@ module.exports = (function(){
     return runInSecond && runOnMinute && runOnHour && runOnDay && runOnMonth;
   }
 
-  function Task(pattern, execution){
+  function Task(pattern, execution, thisArg, args){
     validatePattern(pattern);
     this.initialPattern = pattern.split(' ');
     this.pattern = convertExpression(pattern);
     this.execution = execution;
+    this.arguments = args;
+    this.thisArg = thisArg || execution;
     this.expressions = this.pattern.split(' ');
   }
 
   Task.prototype.update = function(date){
     if(mustRun(this, date)){
       try {
-        this.execution();
+        this.execution.call(this.thisArg, ...this.arguments);
```

Comment:
I believe this codebase is built to run on old versions of node so this line introduces a syntax error.